service-commands/src/createUserWithTrackUpload.js

```diff
@@ -0,0 +1,294 @@
+  const targetFilePath = path.resolve(localDirPath, `${genRandomString(6)}.mp3`)
+
+  const response = await fetch(trackURL)
+  if (!response.ok) {
+    throw new Error(`unexpected response ${response.statusText}`)
+  }
+
+  try {
+    await fs.ensureDir(localDirPath)
+    await streamPipeline(response.body, fs.createWriteStream(targetFilePath))
+
+    console.info(`Wrote track to temp local storage at ${targetFilePath}`)
+  } catch (e) {
+    const errorMsg = `Error with writing track to path ${localDirPath}`
+    console.error(errorMsg, e)
+    throw new Error(`${errorMsg}: ${e.message}`)
+  }
+
+  // Return full file path
+  return targetFilePath
+}
+
+// =~*=~*=~*=~*=~*=~*=~*=~*=~*=~* END UTIL =~*=~*=~*=~*=~*=~*=~*=~*=~*=~*
+const configureAndInitLibs = async () => {
+  const audiusLibsConfig = {
+    ethWeb3Config: AudiusLibs.configEthWeb3(
+      ETH_TOKEN_ADDRESS,
+      ETH_REGISTRY_ADDRESS,
+      ETH_PROVIDER_ENDPOINT,
+      ETH_OWNER_WALLET
+    ),
+    web3Config: AudiusLibs.configInternalWeb3(
+      DATA_CONTRACTS_REGISTRY_ADDRESS,
+      DATA_CONTRACTS_PROVIDER_ENDPOINT
+    ),
+    creatorNodeConfig: AudiusLibs.configCreatorNode(USER_METADATA_ENDPOINT),
+    discoveryProviderConfig: AudiusLibs.configDiscoveryProvider(new Set([DISCOVERY_NODE_ENDPOINT])),
+    identityServiceConfig: AudiusLibs.configIdentityService(IDENTITY_SERVICE_ENDPOINT),
+    isServer: true,
+    isDebug: true
+  }
+
+  const audiusLibs = new AudiusLibs(audiusLibsConfig)
+
+  try {
+    await audiusLibs.init()
+  } catch (e) {
+    console.error('Couldn\'t init libs', e)
+    throw e
+  }
+
+  // Clears local storage in case there is any pre-existing user state
+  audiusLibs.userStateManager.clearUser()
+
+  AudiusLibsInstance = audiusLibs
+
+  return audiusLibs
+}
+
+async function createUser () {
+  userMetadata = getRandomUser()
+  console.log('Creating user with metadata:\n', userMetadata)
+  const { email, password, profilePictureFile, coverPhotoFile } = userMetadata
+  const signUpResp = await AudiusLibsInstance.Account.signUp(
+    email, password, userMetadata/*, profilePictureFile, coverPhotoFile */
+  )
+
+  wait(5000) // wait 5 sec indexing
+
+  // Option 1: Modify user state in userStateManager and then call this
+  // todo: need to update is_creator flag
+  // await AudiusLibsInstance.Account.updateCreatorNodeEndpoint(explicitPrimary)
+
+  // Option 2: c&p updateCreatorNodeEndpoint code and slightly modify it
+  console.log(`Updating replica set with primary ${explicitPrimary}...`)
+  let user = { ...AudiusLibsInstance.userStateManager.getCurrentUser() }
+  await AudiusLibsInstance.creatorNode.setEndpoint(explicitPrimary)
+  const newCreatorNodeEndpoint = CreatorNode.buildEndpoint(
+    explicitPrimary, 
+    CreatorNode.getSecondaries(user.creator_node_endpoint)
+  )
+  user.creator_node_endpoint = newCreatorNodeEndpoint
+  await AudiusLibsInstance.User.updateCreator(user.user_id, user)
+
+  wait(30000) // wait 30 sec indexing and sync(?)
```

Comment:
we should poll here for the creator node endpoint. https://github.com/AudiusProject/audius-protocol/blob/master/libs/src/api/user.js#L588-L598